recipes/default.rb

```diff
@@ -217,20 +216,38 @@
   mode 0750
   variables(
     {
-      :kafka_fqdn => kafka_fqdn,
       :mgm_fqdn => mgm_fqdn,
       :hopsworks_url => hopsworks_url
     }
   )
 end
 
+kafka_fqdn = consul_helper.get_service_fqdn("broker.kafka")
+template "#{node['onlinefs']['etc']}/#{['onlinefs']['kafka']['properties_file']}" do
+  source "onlinefs-kafka.properties.erb"
+  owner node['onlinefs']['user']
+  group node['onlinefs']['group']
+  mode 0750
+  variables(
+    {
+      :kafka_fqdn => kafka_fqdn
+    }
+  )
+end
+
 template "#{node['onlinefs']['etc']}/log4j.properties" do
   source "log4j.properties.erb"
   owner node['onlinefs']['user']
   group node['onlinefs']['group']
   mode 0750
 end
 
+require 'fileutils'
+if !node['onlinefs']['config_dir'].nil?
+  # Copy everything from the provided config_dir to etc overwriting any duplicates
+  FileUtils.cp_r(Dir["#{node['onlinefs']['config_dir']}/*"], node['onlinefs']['etc'])
+end
+
```

Comment:
I suspect this won't work because the ruby code is executed during the compilation phase of the recipe and the template above is executed during the convergence phase (which is executed after).

Reply:
thanks for the feedback, now it should be fixed.